contribute/design/1.x/jiva/2019152019-jiva-autosnap-deletion.md

```diff
@@ -0,0 +1,98 @@
+
+---
+oep-number: draft Auto Snapshot deltion 20190715
+title: Jiva auto snapshot deletion
+authors:
+  - "@utkarshmani1997"
+owners:
+  - "@vishnuitta"
+  - "@payes"
+  - "@utkarshmani1997"
+editor: "@utkarshmani1997"
+creation-date: 2019-07-15
+last-updated: 2019-07-15
+status: provisional
+see-also:
+  - NA
+replaces:
+  - NA
+superseded-by:
+  - NA
+---
+
+ # Jiva automatic snapshot deletion
+
+ ## Table of Contents
+
+ * [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+* [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Delete autogenerated snapshot](#delete-autogenerated-snapshot)
+    * [Implementation Details](#implementation-details)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+
+ ## Summary
+
+ This proposal is aimed for the users who experience very frequent network glitch
+ at there cluster, due to which replica get restarted and autogenerated snapshots
+ files are created. Currently the limit is 512 snapshots, once these no of snapshots
+ created replica won't be able to connect again automatically and requires manual
+ cleanup of snapshot.
+
+ ## Motivation
+
+ The motivation behind this approach is to remove/reduce the efforts of the user
+ for managing the storage layer
+
+ ### Goals
+
+ - Ability to remove autogenerated snapshots automatically in background without
+   any manual intervention or with minimal effort.
+
+ ## Proposal
+ Delete the autogenerated snapshots in jiva in the background
+
+ ### User Stories
+ #### Delete autogenerated snapshot:
+ As a user i want to cleanup the autogenerated snapshots with no or minimal efforts
+
+ ### Implementation Details
+
+ Though there is a working flow of deletion of snapshot via REST Api/ jivactl cli
+ but this requires manual intervention from user i.e, user needs to list the snap
+ shots and delete it one by one via given approaches but if no of snapshots have
+ been grown significantly it will take too much time to cleanup those snapshots.
+ There are following approaches which can be used to implement this feature:
+
+ - Delete the n no of snapshots via cli tool jiva
+     * For exp: jivactl purge snapshot -n=10
+ - Delete snapshots having size = 0 or very less amount of data is there ~ 1-5 G
+ - Run a goroutine in controller to perform cleanup if no of snapshot grows to
+   more than n (maybe 10/20...50)
```

Comment:
Can the above 2 steps be combined together to delete snapshots smartly.

Reply:
From where will n be passed, are we going to keep this value fixed or will it be passed to controller as env.

Reply:
current implementation is based on the third approach.